File: src/main/java/wedding/alba/function/profile/ProfileUpdateRequestDTO.java
package wedding.alba.function.profile;

import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ProfileUpdateRequestDTO {
    private String nickname;

    private String selfIntroduction;

    private String activityArea;

    private String profileImageUrl;
    
    // 갤러리 이미지 관련 필드
    private List<Long> deleteGalleryImageIds; // 삭제할 갤러리 이미지 ID 목록
    private List<String> newGalleryImages; // 새로 추가할 갤러리 이미지 URL 목록 (Base64 또는 임시 URL)
}

File: frontend/src/components/profile/dto/ProfileUpdateRequestDTO.ts
// ProfileUpdateRequestDTO.ts - 프로필 수정 요청 데이터 타입 정의

export interface ProfileUpdateRequestDTO {
  nickname: string;            // 닉네임 (필수)
  selfIntroduction?: string;   // 자기소개 (선택)
  activityArea?: string;       // 활동 지역 (선택)
  profileImageUrl?: string;    // 프로필 이미지 URL (선택)
}
